rpc/provider/smoldot.ts

```diff
@@ -92,7 +92,14 @@ async function connection(
   }
   let conn = connections.get(chainSpec);
   if (!conn) {
-    const inner = await client.addChain({ chainSpec });
+    const innerMap: Chain[] = [];
+    connections.forEach((value: SmoldotProviderConnection) => {
+      innerMap.push(value.inner);
+    });
+    const inner = await client.addChain({
+      chainSpec,
+      potentialRelayChains: innerMap,
+    });
```

Comment:
Thanks for the suggestion @wirednkod 
This `innerMap` will be re-created when there is no existing connection for the given chainSpec, so it seems that it will always be empty.

Anyway, based on your previous tips, I added
https://github.com/paritytech/capi/pull/374/files#diff-312ccecf597532308900880045fdbe77f2bf18934717d547439a320860f3dcc8R94-R104